# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: modules/libpref/Preferences.cpp
# Commit: a3fd24f5f0a7
# Full Hash: a3fd24f5f0a7bc094409fdbdbe1ef92fc459677b
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-11-23 21:49:00
# Regressor Bug: 1579367
# File Overlap Count: 1
# Description:
#   Bug 1579367 - Initialize XPCJSContext explicitly, after loading user prefs. r=kmag
#   
#   This way we get the correct values for start-up prefs in the parent process.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D51061
# ==============================================================================

diff -r e06f2a4de969 -r a3fd24f5f0a7 modules/libpref/Preferences.cpp
--- a/modules/libpref/Preferences.cpp	Fri Nov 22 09:48:58 2019 +0000
+++ b/modules/libpref/Preferences.cpp	Fri Nov 22 12:40:17 2019 +0000
@@ -3668,7 +3668,10 @@
 
   // Don't set mCurrentFile until we're done so that dirty flags work properly.
   sPreferences->mCurrentFile = prefsFile.forget();
-
+}
+
+/* static */
+void Preferences::FinishInitializingUserPrefs() {
   sPreferences->NotifyServiceObservers(NS_PREFSERVICE_READ_TOPIC_ID);
 
   // At this point all the prefs files have been read and telemetry has been